EXILED/Exiled.Events/Patches/Events/Server/SelectingRespawnTeam.cs

```diff
@@ -27,35 +27,43 @@ namespace Exiled.Events.Patches.Events.Server
     /// Patches <see cref="WaveManager"/> to add the <see cref="Handlers.Server.SelectingRespawnTeam"/> event.
     /// </summary>
     [EventPatch(typeof(Handlers.Server), nameof(Handlers.Server.SelectingRespawnTeam))]
-
-    // [HarmonyPatch(typeof(WaveManager), nameof(RespawnManager.Update))] TODO Idk which method to patch
+    [HarmonyPatch(typeof(WaveManager), nameof(WaveManager.InitiateRespawn))]
     internal static class SelectingRespawnTeam
     {
         private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
             List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
             LocalBuilder ev = generator.DeclareLocal(typeof(SelectingRespawnTeamEventArgs));
 
-            const int offset = 1;
-            int index = newInstructions.FindLastIndex(i => i.opcode == OpCodes.Stloc_1) + offset;
+            Label returnLabel = generator.DefineLabel();
+
+            int index = newInstructions.FindIndex(i => i.opcode == OpCodes.Ldc_I4_1);
 
             newInstructions.InsertRange(index, new[]
             {
                 // SelectingRespawnTeamEventArgs ev = new(dominatingTeam);
-                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsTo(newInstructions[index]),
                 new(OpCodes.Newobj, GetDeclaredConstructors(typeof(SelectingRespawnTeamEventArgs))[0]),
                 new(OpCodes.Dup),
+                new(OpCodes.Dup),
                 new(OpCodes.Stloc_S, ev),
 
                 // Handlers.Server.OnSelectingRespawnTeam(ev);
                 new(OpCodes.Call, Method(typeof(Handlers.Server), nameof(Handlers.Server.OnSelectingRespawnTeam))),
 
+                // if (!ev.IsAllowed)
+                //     return;
+                new(OpCodes.Callvirt, PropertyGetter(typeof(SelectingRespawnTeamEventArgs), nameof(SelectingRespawnTeamEventArgs.IsAllowed))),
+                new(OpCodes.Brfalse_S, returnLabel),
+
                 // dominatingTeam = ev.Team;
-                new(OpCodes.Ldloc, ev),
-                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(SelectingRespawnTeamEventArgs), nameof(SelectingRespawnTeamEventArgs.Team))),
-                new(OpCodes.Stloc_1),
+                new(OpCodes.Ldloc_S, ev),
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(SelectingRespawnTeamEventArgs), nameof(SelectingRespawnTeamEventArgs.Wave))),
+                new(OpCodes.Starg_S, 0),
```

Comment:
```suggestion
                new(OpCodes.Starg_S, 1),
```

I haven't looked at the base game code but it would be 0 if you are changing the instance itself.

Reply:
it's a static class